openapi.yaml

```diff
@@ -2980,26 +2980,27 @@ paths:
     get:
       summary: Download results for a completed batch job
       operationId: list-results
-      description: >-
+      description: |-
         Lists signed URLs pointing to the processed files after the batch job
         has finished. Back-ends may also point to intermediate results after the
         job has stopped due to an error.
-        
 
         The response includes additional metadata. It is a valid
-        [STAC Item (version 0.9.0)](https://github.com/radiantearth/stac-spec/tree/v0.9.0/item-spec)
+        [STAC Item](https://github.com/radiantearth/stac-spec/tree/v0.9.0/item-spec)
         if it has spatial and temporal references included.
 
-
         URL signing is a way to protect files from unauthorized access with a
         key in the URL instead of HTTP header based authorization. The URL
         signing key is similar to a password and its inclusion in the URL allows
         to download files using simple GET requests supported by a wide range of
         programs, e.g. web browsers or download managers. Back-ends are
         responsible to generate the URL signing keys and to manage their
         appropriate expiration. The back-end MAY indicate an expiration time by
-        setting the `expires` property.
-
+        setting the `expires` property in the reponse. Requesting this endpoint
+        SHOULD always return valid URLs. If the links generated for a previous
+        request have expired, the signed URLs SHOULD be renewed. 
+        Once the signed URLs have expired, the signed URLs MAY return the openEO
+        error `LinkExpired`.
```

Comment:
It took me some re-reading to fully understand the points you're making here.

fine-tuning suggestion:

> Requesting this endpoint SHOULD always return URLs that are valid within a reasonable time window.
> Signed URLs that were generated for a previous request and already expired SHOULD NOT be reused, but regenerated with new expiration time.
> Signed URLs that expired MAY return the openEO error `LinkExpired`.

Reply:
Mostly adopted your proposal.